src/geventhttpclient/useragent.py

```diff
@@ -314,6 +321,90 @@ def _verify_status(self, status_code, url=None):
         if status_code not in self.valid_response_codes:
             raise BadStatusCode(url, code=status_code)
 
+    def _encode_files(self, files, data):
+        """
+        Method taken from models in requests library , usage is the same. Only difference is that you can add custom
+        boundary in 5-tuple version.
+
+        Build the body for a multipart/form-data request.
+
+        Will successfully encode files when passed as a dict or a list of
+        tuples. Order is retained if data is a list of tuples but arbitrary
+        if parameters are supplied as a dict.
+
+        The tuples may be
+        2-tuples (filename, fileobj),
+        3-tuples (filename, fileobj, contentype),
+        4-tuples (filename, fileobj, contentype, custom_headers) or
+        5-tuples (filename, fileobj, contentype, custom_headers, custom boundary).
+
+        example:
+        files = {'file': ('report.xls', body, 'application/vnd.ms-excel', {'Expires': '0'}, 'custom_boundary')}
```

Comment:
@flyingbutter hey, thanks for the great contribution.
Why are `custom_boundary` specified as the fifth argument to each of the form-data elements? Shouldn't `custom_boundary` be specified only 1 time for all form-data elements? What will be the behavior if you specify different `custom_boundary` in different elements?

Reply:
I didnt want to change the usage of files parameter, it is the same as requests library, so I added custom boundary as 5 tuple type. I think it will use the last custom boundary for all items since they are all encoded in same method.